### PR TITLE
Variational Inference uses floatX

### PR DESCRIPTION
**Pull request by ferrine**

### CODE REVIEW

**Comment by ferrine** [comment]
Let's merge?

**Comment by twiecki** [comment]
Just taking a look.

File: pymc3/distributions/dist_math.py [hunk 1]
```diff
@@ -156,9 +157,9 @@ def log_normal(x, mean, **kwargs):
     elif rho is not None:
         std = rho2sd(rho)
     else:
-        std = tau**(-1)
+        std = tau**(f(-1))
     std += eps
-    return c - tt.log(tt.abs_(std)) - (x - mean) ** 2 / (2 * std ** 2)
+    return f(c) - tt.log(tt.abs_(std)) - (x - mean) ** f(2) / (f(2) * std ** f(2))
```
Comment (twiecki) [inline pymc3/distributions/dist_math.py]: Is that really necessary? I think python floats are not the problem, only numpy.

File: pymc3/tests/conftest.py [hunk 2]
```diff
@@ -7,3 +7,12 @@ def theano_config():
     config = theano.configparser.change_flags(compute_test_value='raise')
     with config:
         yield
+
+
+@pytest.fixture(scope='function')
```
Comment (twiecki) [inline pymc3/tests/conftest.py:12]: neat!

**Comment by twiecki** [comment]
Yeah, mainly I wonder if all that casting is really necessary, can you try removing some scalar conversions that are not numpy?

File: pymc3/distributions/dist_math.py (hunk 1 above)
Comment (ferrine) [reply pymc3/distributions/dist_math.py]: I got an exception right here

**Comment by ferrine** [comment]
Some of them are not so important. I can try but I see no profits

**Comment by twiecki** [comment]
Profit is code simplicity / readability.

File: pymc3/distributions/dist_math.py (hunk 1 above)
Comment (twiecki) [reply pymc3/distributions/dist_math.py]: which did you remove? e.g. ** f(2), i'd be surprised if that's required.

Comment (junpenglao) [reply pymc3/distributions/dist_math.py]: maybe `f(2)` -> `2.` ?

**Comment by twiecki** [comment]
Awesome!

**Pull request merged**